fili-core/src/main/java/com/yahoo/bard/webservice/web/responseprocessors/DruidPartialDataResponseProcessor.java

```diff
@@ -0,0 +1,257 @@
+     */
+    private void checkOverflow(JsonNode json, DruidAggregationQuery<?> query) {
+        if (json.get(DruidJsonResponseContentKeys.DRUID_RESPONSE_CONTEXT.getName())
+                .get(DruidJsonResponseContentKeys.UNCOVERED_INTERVALS_OVERFLOWED.getName())
+                .asBoolean()
+    ) {
+            logAndGetErrorCallback(
+                    ErrorMessageFormat.TOO_MANY_INTERVALS_MISSING.format(
+                            query.getContext().getUncoveredIntervalsLimit()
+                    ),
+                    query
+            );
+        }
+    }
+
+    /**
+     * Logs and gets error call back on the response with the provided error message.
+     *
+     * @param message  The error message passed to the logger and the exception
+     * @param query  The query with the schema for processing this response
+     */
+    private void logAndGetErrorCallback(String message, DruidAggregationQuery<?> query) {
+        LOG.error(message);
+        getErrorCallback(query).dispatch(
+                Status.INTERNAL_SERVER_ERROR.getStatusCode(),
+                "The server encountered an unexpected condition which prevented it from fulfilling the request.",
+                message);
+    }
+
+    /**
+     * Returns the overlap between uncoveredIntervals from Druid and missing intervals that Fili expects.
+     *
+     * @param json  The JSON node that contains the uncoveredIntervals from Druid, for example
+     * <pre>
+     * {@code
+     * X-Druid-Response-Context: {
+     *     "uncoveredIntervals": [
+     *         "2016-11-22T00:00:00.000Z/2016-12-18T00:00:00.000Z","2016-12-25T00:00:00.000Z/2017-
+     *         01-03T00:00:00.000Z","2017-01-31T00:00:00.000Z/2017-02-01T00:00:00.000Z","2017-02-
+     *         08T00:00:00.000Z/2017-02-09T00:00:00.000Z","2017-02-10T00:00:00.000Z/2017-02-
+     *         13T00:00:00.000Z","2017-02-16T00:00:00.000Z/2017-02-20T00:00:00.000Z","2017-02-
+     *         22T00:00:00.000Z/2017-02-25T00:00:00.000Z","2017-02-26T00:00:00.000Z/2017-03-
+     *         01T00:00:00.000Z","2017-03-04T00:00:00.000Z/2017-03-05T00:00:00.000Z","2017-03-
+     *         08T00:00:00.000Z/2017-03-09T00:00:00.000Z"
+     *     ],
+     *     "uncoveredIntervalsOverflowed": true
+     * }
+     * }
+     * </pre>
+     * @param query  The Druid query that contains the missing intervals that Fili expects
+     *
+     * @return the overlap between uncoveredIntervals from Druid and missing intervals that Fili expects.
+     */
+    private SimplifiedIntervalList getOverlap(JsonNode json, DruidAggregationQuery<?> query) {
+        List<Interval> intervals = new ArrayList<>();
+        for (JsonNode jsonNode :
+                json.get(DruidJsonResponseContentKeys.DRUID_RESPONSE_CONTEXT.getName())
+                        .get(DruidJsonResponseContentKeys.UNCOVERED_INTERVALS.getName())
+                ) {
+            intervals.add(new Interval(jsonNode.asText()));
```

Comment:
If it is a ObjectNode, it is some kind of key value mapping right? So what is the key for each individual string like `"2016-11-22T00:00:00.000Z/2016-12-18T00:00:00.000Z"`?